cmd/agent/common/misconfig/global.go

```diff
@@ -8,19 +8,25 @@ package misconfig
 import "github.com/DataDog/datadog-agent/pkg/util/log"
 
 // ToLog outputs warnings about common misconfigurations in the logs
-func ToLog() {
-	for name, check := range checks {
-		if err := check(); err != nil {
-			log.Warnf("misconfig: %s: %v", name, err)
+func ToLog(agent string) {
+	for _, check := range checks {
+		if check.Agent == agent {
+			if err := check.Run(); err != nil {
+				log.Warnf("misconfig: %s: %v", agent, err)
+			}
 		}
 	}
 }
 
 type checkFn func() error
+type check struct {
+	Agent string
+	Run   checkFn
```

Comment:
nit: this probably should be unexported since the `check` struct isn't used outside of the `misconfig` package